src/js/actions/layers.js

```diff
@@ -1377,45 +1382,147 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard 
+     * @param {Immutable.List.of<Layer>} artboards
+     * @return {Layer}
+     */
+
+    var _getRightMost = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+                if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                    return currentLayer;
+                } else {
+                    return selectedLayer;
+                }
+            }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     * @param {string} specs, {Immutable.List.of<Layer>} artboards, {Layer} layerRef
+     * @return {Bounds} finalBounds
+     */
+
+    var _getBoundsFromSpecs = function (specs, artboards, layerRef) {
+        var preset,
+            width, 
+            height,
+            rightmostlayer,
+            finalBounds;
+
+        //get the preset template from the specs
+        preset = specs.preset;
+        //find the measurements of that preset 
+        _.each(templates, function (value, index) { 
```

Comment:
It would be better to use something like `_.findIndex` for this so that a) you don't have to iterate over the entire array unnecessarily, and b) you can cleanly separate the search from what you're going to do with the search result.